cms/djangoapps/contentstore/views/program.py

```diff
@@ -38,3 +39,16 @@ def get(self, request, *args, **kwargs):
             })
         else:
             raise Http404
+
+
+class ProgramsIdTokenView(View):
+    """Provides id tokens to JavaScript clients for use with the Programs API."""
+
+    @method_decorator(login_required)
```

Comment:
The test passes.  Since it only implements GET, does it matter?

On Monday, December 7, 2015, Renzo Lucioni <notifications@github.com> wrote:

> In cms/djangoapps/contentstore/views/program.py
> <https://github.com/edx/edx-platform/pull/10862#discussion_r46849795>:
>
> > @@ -38,3 +39,16 @@ def get(self, request, *args, **kwargs):
> >              })
> >          else:
> >              raise Http404
> > +
> > +
> > +class ProgramsIdTokenView(View):
> > +    """Provides id tokens to JavaScript clients for use with the Programs API."""
> > +
> > +    @method_decorator(login_required)
>
> Does this work? Django's docs on class-based views recommend decorating
> the dispatch method
> <https://docs.djangoproject.com/en/1.9/topics/class-based-views/intro/#decorating-the-class>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/10862/files#r46849795>.
>